File: LondonFhirService.Manage.Client/src/models/foundations/metrics/Metric.ts
// Foundation model - mirrors LondonFhirService.Core.Models.Foundations.Metrics.Metric as the
// /api/metrics endpoint serialises it.
//
// type and status arrive as ORDINALS, not names. The Manage host registers no
// JsonStringEnumConverter, so although EF persists both as text, on the wire they are numbers.
export type Metric = {
    id: string;
    userId: string | null;
    parentId: string | null;
    correlationId: string;
    method: string;
    type: number;
    name: string;
    target: string | null;
    started: string;
    completed: string;
    durationMs: number;
    status: number;
    errorCode: string | null;
    payloadBytes: number | null;
    consumer: string | null;
    description: string | null;
    createdDate: string;
};

File: LondonFhirService.Manage.Tests.Acceptance/Models/Metrics/Metric.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace LondonFhirService.Manage.Tests.Acceptance.Models.Metrics
{
    /// <summary>
    /// Type and Status are strings rather than the enums Core uses. The broker persists both as
    /// text so the reporting queries stay readable, and the acceptance suite talks to the wire
    /// format rather than to Core's types.
    /// </summary>
    public class Metric
    {
        public Guid Id { get; set; }
        public Guid? ParentId { get; set; }
        public Guid CorrelationId { get; set; }
        public string Method { get; set; }
        public string Type { get; set; }
        public string Name { get; set; }
        public string Target { get; set; }
        public DateTimeOffset Started { get; set; }
        public DateTimeOffset Completed { get; set; }
        public double DurationMs { get; set; }
        public string Status { get; set; }
        public string ErrorCode { get; set; }
        public long? PayloadBytes { get; set; }
        public string Consumer { get; set; }
        public string Description { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
    }
}
